packages/flutter_test/lib/src/binding.dart

```diff
@@ -700,13 +729,33 @@ abstract class TestWidgetsFlutterBinding extends BindingBase
     // The LiveTestWidgetsFlutterBinding overrides this to report the exception to the console.
   }
 
+  Future<dynamic> _handleMessage(dynamic mockMessage) async {
+    final Map<dynamic, dynamic> data =
+        (mockMessage as Map<dynamic, dynamic>)['data'] as Map<dynamic, dynamic>;
+    final String message = data['message'] as String;
+    final TextDirection textDirection = TextDirection.values[data['textDirection'] as int];
+    final int assertivenessLevel = (data['assertiveness'] as int?) ?? 0;
+    final Assertiveness assertiveness = Assertiveness.values[assertivenessLevel];
+    final CapturedAccessibilityAnnouncement announcement;
+
+    announcement = CapturedAccessibilityAnnouncement(message, textDirection, assertiveness: assertiveness);
+    _announcements ??= <CapturedAccessibilityAnnouncement>[];
+    _announcements!.add(announcement);
+  }
+
   Future<void> _runTest(
     Future<void> Function() testBody,
     VoidCallback invariantTester,
     String description,
   ) {
     assert(description != null);
     assert(inTest);
+
+    TestDefaultBinaryMessengerBinding.instance!.defaultBinaryMessenger
```

Comment:
In stead of hardcode this in every test, can you provide an adhoc API?

For example user can do something like
```dart
testWidget('test', (WidgetTester tester) {
  var capturer = tester.captureAnnouncement();
  expect(capturer.takeAnnouncement, ...)
  capturer.dispose();
})
```

The problem with this approach is this may override the entire accessibility channel,  so it may not always be preferred.

Reply:
You still get the option to have a custom accessibility channel (there's even a test covering this use-case). Is there an example scenario you are worried about?

Reply:
That sounds fine then.